### ISSUE TITLE
@isModified doesn't work in sub-docs if they created throuth the #create method

### ISSUE THREAD

**Issue by lazutkin** [Automattic/mongoose#1814]
Sub-docs doesn't track changes (`mongoose 3.8.1`)

```coffee
mongoose = require "mongoose"
Schema = mongoose.Schema

Car = new Schema {
  color: String
  model: String
}

Person = new Schema {
  name: String
  cars: [ Car ]
}

Person = mongoose.model "Person", Person

mongoose.connect "mongodb://localhost/test"
db = mongoose.connection
db.on "open", ->

  vasya = new Person { name: "Vasya Pupkin" }
  car   = vasya.cars.create { model: "VAZ 2101", color: "black" }

  console.log "isNew = #{ car.isNew }"
  console.log "selected = #{ car.isSelected "model" }"
  console.log "modified = #{ car.isModified "model" }, value = #{ car.model }"

```

```
isNew = true
selected = true
modified = false, value = VAZ 2101
```

**Comment by dongliu** [comment]
You might want to use the #isNew to check in such cases.

**Comment by lazutkin** [comment]
@dongliu and I doing so, but this is workaround of a bug.

**Comment by vkarpov15** [comment]
Will check this out, thanks for the reminder. Sorry this slipped through the cracks

**Comment by vkarpov15** [comment]
This bug has been fixed since 4.2.5:

```javascript
'use strict';

Error.stackTraceLimit = Infinity;

var assert = require('assert');
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

mongoose.connect('mongodb://localhost/gh1814');
mongoose.set('debug', true);

var Car, Person, Schema, db;

Car = new Schema({
  color: String,
  model: String
});

Person = new Schema({
  name: String,
  cars: [Car]
});

Person = mongoose.model("Person", Person);

db = mongoose.connection;

db.on("open", function() {
  var car, vasya;
  vasya = new Person({
    name: "Vasya Pupkin"
  });
  car = vasya.cars.create({
    model: "VAZ 2101",
    color: "black"
  });
  console.log("isNew = " + car.isNew);
  console.log("selected = " + (car.isSelected("model")));
  return console.log("modified = " + (car.isModified("model")) + ", value = " + car.model);
});
```

```
$ rm -rf node_modules/mongoose && npm install mongoose@4.2.5

$ node gh-1814.js 
isNew = true
selected = true
modified = true, value = VAZ 2101
^C
$ 
```
